Examples/PcapSplitter/IPPortSplitters.h

```diff
@@ -447,3 +497,76 @@ class ServerPortSplitter : public IPPortSplitter
 		return prefix + sstream.str();
 	}
 };
+
+/**
+ * Splits a pcap file by server client (most of the time is similar to protocol). This means that all flows with a certain
+ * client port will be written to the same file. The client port for each flow is determined as follows: 1) if it's a TCP
+ * flow and we have the SYN packet - the client port is the source port of the SYN packet 2) if it's a TCP flow and we only
+ * have the SYN/ACK packet - the client port is the dest port of the SYN/ACK packet 3) if it's a partial TCP flow and
+ * we don't have the SYN or SYN/ACK packets, the server port will be determined by the port: the higher port is considered
+ * the client side 4) if it's a UDP multicast flow - if the sourceIP is a multicast address, the source port is considered
+ * as a client port, otherwise if the destIP is a multicast address, the dest port is considered as a client port 5) if
+ * it's a UDP flow - the client port will be determined by the port: the higher port is considered as client port
+ */
+class ClientPortSplitter : public IPPortSplitter

```

Comment:
I'm not sure how useful this would be for users, that's why I didn't implement this in the first place. Server port is probably more useful. What do you think?

I'll leave it to you to decide because I don't have a hard opinion about this. However, if you decide to include it, would it be possible to add tests for it? Basically, you can add a test to [`test_pcapsplitter.py`](https://github.com/seladb/PcapPlusPlus/blob/master/Tests/ExamplesTest/tests/test_pcapsplitter.py)

Reply:
As an example for us it make sense as we want to analyse multicast feed by destination port which is in this case is the client port. 

I added a test for client port splitting.